### PR TITLE
Internationalization - Application Terms of Use

### PR DESCRIPTION
**Pull request by JayanthyChengan**
## Related Issues

- https://github.com/IQSS/dataverse/issues/5303

### CODE REVIEW

**Comment by scolapasta** [comment]
@JayanthyChengan overall this looks good.

I think we should not have EN for lang for all settings, especially since not all settings are user text, so blank lang, which is default, but then you can add language specific ones if you need. So the logic should be if you don't provide a lang, it returns the default (and most settings would do that); if you do provide a language it looks for that language, but if not available returns the default.

**Comment by JayanthyChengan** [comment]
@scolapasta  - Thanks for your review.  I would like to add some notes of what I tried.
As a first step, I added new column “lang” and when a new row is added(:ApplicationTermsOfUse , "french content" , "fr" ) , it throws the below exception,

INSERT INTO setting (name, content, lang)
VALUES
(:ApplicationTermsOfUse , "french content" , "fr" );
 
**org.postgresql.util.PSQLException: ERROR: **duplicate key value** violates unique constraint "setting_pkey"**

 
So, I added “lang” to the primary key (name,lang), and  updated to  default value “en” as primary key cannot be NULL
 
https://github.com/IQSS/dataverse/blob/b9bed069c01a5587cc1a75c551ff4352dd9fc755/scripts/database/upgrades/upgrade_v4.15.1_to.sql
 
setting_pkey PRIMARY KEY (name,lang)
 
Please add your suggestion.

**Comment by JayanthyChengan** [comment]
@scolapasta - Thanks Gustavo and I implemented the suggested solution using coalesce, it worked.  Committed the code,  please review.  Thanks

File: src/main/java/propertyFiles/Bundle.properties [hunk 1]
```diff
@@ -2383,6 +2383,7 @@ staticSearchFields.dvObjectType=Type
 staticSearchFields.fileTag=File Tag
 staticSearchFields.fileAccess=Access
 staticSearchFields.publicationStatus=Publication Status
+staticSearchFields.subject_ss=Subject
```
Comment (scolapasta) [inline src/main/java/propertyFiles/Bundle.properties]: @JayanthyChengan is this supposed to be part of this PR?

File: scripts/database/upgrades/upgrade_v4.15.1_to.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE ONLY setting DROP CONSTRAINT setting_pkey ;
+
+ALTER TABLE setting ADD COLUMN ID SERIAL PRIMARY KEY;
```
Comment (scolapasta) [inline scripts/database/upgrades/upgrade_v4.15.1_to.sql]: does this add values for the id for existing settings? (it must, right?)

Also, can you update the file name to reference 4.16 (the latest release?)

Comment (JayanthyChengan) [reply scripts/database/upgrades/upgrade_v4.15.1_to.sql]: Yes, it adds serial number to all existing rows.  I will rename the file and commit it.

File: src/main/java/propertyFiles/Bundle.properties (hunk 1 above)
Comment (JayanthyChengan) [reply src/main/java/propertyFiles/Bundle.properties]: @scolapasta  - Missing term related to facet Internationalization.  hope it's okay to add with this PR.

File: scripts/database/upgrades/upgrade_v4.15.1_to.sql (hunk 2 above)
Comment (JayanthyChengan) [reply scripts/database/upgrades/upgrade_v4.15.1_to.sql]: committed the changes.

**Comment by kcondon** [comment]
@JayanthyChengan Would you refresh from develop? The app version is still 4.15.1.

Thanks!

**Comment by JayanthyChengan** [comment]
@kcondon - merged the latest develop branch.

**Comment by kcondon** [comment]
@JayanthyChengan thanks. I am seeing a 500 error when I load the app in a browser with a sql error in logs:
Call: SELECT ID, CONTENT, LANG, NAME FROM SETTING WHERE ((NAME = ?) AND (LANG IS NULL))
        bind => [1 parameter bound]
Query: ReadAllQuery(name="Setting.findByName" referenceClass=Setting sql="SELECT ID, CONTENT, LANG, NAME FROM SETTING WHERE ((NAME = ?) AND (LANG IS NULL))")

**Comment by JayanthyChengan** [comment]
Hi @kcondon - there is a setting table update, I am not sure whether you executed the db update script?  Let me know if the error still exists.  Thanks
 https://github.com/IQSS/dataverse/blob/f199fce7b3146877d9576bb474fcd3f3186cdfd7/scripts/database/upgrades/upgrade_v4.16_to_4.16.1.sql

**Comment by kcondon** [comment]
@JayanthyChengan We now use an automatic db update management s/w called Flyway. We no longer use the updates dir method. If you would use the Flyway method, that would be great: http://guides.dataverse.org/en/4.16/developers/sql-upgrade-scripts.html# Thanks!

**Comment by JayanthyChengan** [comment]
@kcondon  - Added the db upgrade file as per Flyway method.  Please verify it now.  Thanks

**Comment by kcondon** [comment]
@JayanthyChengan Almost there! The new script needs to be renamed: you need to use a capital V rather a v at the beginning of the script name, otherwise flyway ignores the script. Annoying, I know.

**Comment by JayanthyChengan** [comment]
@kcondon - got it, and renamed the file .

**Pull request merged**